app/src/main/java/explorer/city/com/cityexplorer/view/adapter/viewHolder/DetailsViewHolder.kt

```diff
@@ -0,0 +1,56 @@
+package explorer.city.com.cityexplorer.view.adapter.viewHolder
+
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import explorer.city.com.cityexplorer.model.CityDetail
+import explorer.city.com.cityexplorer.util.DetailType
+import kotlinx.android.synthetic.main.list_item__detail.view.*
+
+class DetailsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+    fun setDetail(detail: CityDetail) {
+        when(detail.type) {
+            DetailType.CURRENCY.type -> renderCurrency(detail)
+            DetailType.FLOAT.type -> renderFloat(detail)
+            DetailType.INT.type -> renderInt(detail)
+            DetailType.PERCENT.type -> renderPercent(detail)
+            DetailType.URL.type -> renderUrl(detail)
+            else -> renderString(detail)
+        }
+    }
+
+    private fun renderCurrency(detail: CityDetail) {
+        val format = if (detail.currencyDollarValue > 10) "%.0f$" else "%.1f$"
+        val score = String.format("%s: $format", detail.label, detail.currencyDollarValue)
+        setText(score)
+    }
+
+    private fun renderFloat(detail: CityDetail) {
+        val score = String.format("%s: %.2f", detail.label, detail.floatValue)
+        setText(score)
+    }
+
+    private fun renderInt(detail: CityDetail) {
+        val score = String.format("%s: %.0f", detail.label, detail.intValue)
+        setText(score)
+    }
+
+    private fun renderPercent(detail: CityDetail) {
+        val score = String.format("%s: %f", detail.label, detail.percentValue)
+        setText(score)
+    }
+
+    private fun renderUrl(detail: CityDetail) {
+        val score = String.format("%s: %s", detail.label, detail.urlValue)
+        setText(score)
+    }
+
+    private fun renderString(detail: CityDetail) {
+        val score = String.format("%s: %s", detail.label, detail.stringValue)
+        setText(score)
```

Comment:
This is less than ideal. It would be better for all of these function to return a String; then you would only need one call to `setText`. It's not a huge problem. The risk is that, when a new function is added such as `renderBigInteger` the developer needs to remember to also call `setText`.

Your solution also adds more lines of code to this class.